paws_collector.js

```diff
@@ -780,6 +808,79 @@ class PawsCollector extends AlAwsCollector {
         });
     }
 
+    /**
+     * Get hash of message 
+     * @param {*} message 
+     * @returns 
+     */
+    getHash(message) {
+        const hashValue = crypto.createHash('sha256').update(JSON.stringify(message, Object.keys(message).sort())).digest('hex');
+        return hashValue;
+    }
+
+   /**
+    * 
+    * @param {*} logs 
+    * @param {*} paramName :Uniquely identified parameter key 
+    * @param {*} callback 
+    * @returns 
+    */
+    removeDuplicatedItem(logs, paramName, asyncCallback) {
+        let collector = this;
+        const ddb = new AWS.DynamoDB();
+        let uniqueLogs = [];
+        var promises = [];
+        let duplicateCount = 0;
+        logs.forEach(record => {
+            const messageHashId = collector.getHash(record);
+            const params = {
+                Item: {
+                    Id: { S: record[`${paramName}`] },
+                    CollectorId: { S: collector._collectorId },
+                    MsgHashId: { S: messageHashId },
+                    // setting DDB time to life. This is set to cover 24hr duplication window
+                    ExpireDate: { N: moment().add(1, 'days').unix().toString() }
+                },
+                TableName: collector._pawsDeDupLogsDdbTableName,
+                ConditionExpression: '(attribute_not_exists(Id) AND attribute_not_exists(MsgHashId)) OR  (Id = :id AND MsgHashId <> :msg_hash_id AND CollectorId = :collector_id)', // Check if item with same Id already exists and id and message is identical
```

Comment:
can we construct `Id` from `cid`, `collectorId`, and `messageHashId` and simplify the condition expression? the combination of cid + collector ID + messageHash should be enough to uniquely identify a message. we can still store other data for operational purposes though. 

maybe something like this?

```javascript
// ...
const cid = getCustomerId();  // I don't know where we usually take CID from 😅
const collectorId = collector._collectorId;
const messageHash = getHash(message);
const itemId = `${cid}_${collectorId}_${messageHash}`;
const params = {
  Item: {
    Id: { S: itemId },
    CollectorId: { S: collectorId },
    OrigMessageId: { S: record[`${paramName}`] },
    ExpireDate: { N: calculateExpiryTs() },
    TableName: collector._pawsDedDupLogsTableName,
    ConditionExpression: 'attribute_not_exists(Id)'
  }
};
// ...
```